File: openapi-generator/scalaz/src/main/scala/org/openapitools/client/api/VehicleMaintenancePassenger.scala
package org.openapitools.client.api

import argonaut._
import argonaut.EncodeJson._
import argonaut.DecodeJson._

import org.http4s.{EntityDecoder, EntityEncoder}
import org.http4s.argonaut._
import org.joda.time.DateTime
import VehicleMaintenancePassenger._

case class VehicleMaintenancePassenger (
  checkEngineLight: Option[VehicleMaintenancePassengerCheckEngineLight],
/* Passenger vehicle DTCs. */
  diagnosticTroubleCodes: Option[List[VehicleMaintenancePassengerDiagnosticTroubleCodes]])

object VehicleMaintenancePassenger {
  import DateTimeCodecs._

  implicit val VehicleMaintenancePassengerCodecJson: CodecJson[VehicleMaintenancePassenger] = CodecJson.derive[VehicleMaintenancePassenger]
  implicit val VehicleMaintenancePassengerDecoder: EntityDecoder[VehicleMaintenancePassenger] = jsonOf[VehicleMaintenancePassenger]
  implicit val VehicleMaintenancePassengerEncoder: EntityEncoder[VehicleMaintenancePassenger] = jsonEncoderOf[VehicleMaintenancePassenger]
}

File: openapi-generator/javascript-closure-angular/API/Client/VehicleMaintenancePassenger.js
goog.provide('API.Client.VehicleMaintenance_passenger');

/**
 * Passenger vehicle data. Null if no data is available.
 * @record
 */
API.Client.VehicleMaintenancePassenger = function() {}

/**
 * @type {!API.Client.VehicleMaintenance_passenger_checkEngineLight}
 * @export
 */
API.Client.VehicleMaintenancePassenger.prototype.checkEngineLight;

/**
 * Passenger vehicle DTCs.
 * @type {!Array<!API.Client.VehicleMaintenance_passenger_diagnosticTroubleCodes>}
 * @export
 */
API.Client.VehicleMaintenancePassenger.prototype.diagnosticTroubleCodes;

